micro_dl/inference/stitch_predictions.py

```diff
@@ -0,0 +1,261 @@
+"""Stich model predictions either along Z or as tiles"""
+import numpy as np
+
+
+class ImageStitcher:
+    """Stitch prediction images for 3D
+
+    USE PREDICT ON LARGER IMAGE FOR 2D AND 2.5D
+    """
+
+    def __init__(self, tile_option,
+                 overlap_dict,
+                 image_format='zyx',
+                 data_format='channels_first'):
+        """Init
+
+        :param str tile_option: 'tile_z' or 'tile_xyz'
+        :param dict overlap_dict: with keys overlap_shape, overlap_operation.
+         overlap_shape is an int for tile_z and list of len 3 for tile_xyz.
+        :param str image_format: xyz or zyx
+        :param str data_format: channels_first or channels_last
+        """
+
+        assert tile_option in ['tile_z', 'tile_xyz'], \
+            'tile_option not in [tile_z, tile_xyz]'
+
+        allowed_overlap_opn = ['mean', 'any']
+        assert ('overlap_operation' in overlap_dict and
+                overlap_dict['overlap_operation'] in allowed_overlap_opn), \
+            'overlap_operation not provided or not in [mean, any]'
+        assert image_format in ['zyx', 'xyz'], 'image_format not in [zyx, xyz]'
+
+        self.tile_option = tile_option
+        self.overlap_dict = overlap_dict
+        self.data_format = data_format
+
+        img_dim = [2, 3, 4] if self.data_format == 'channels_first' \
+            else [1, 2, 3]
+        self.img_dim = img_dim
+        if data_format == 'channels_first':
+            x_dim = 4 if image_format == 'zyx' else 2
+            z_dim = 2 if image_format == 'zyx' else 4
+            y_dim = 3
+        elif data_format == 'channels_last':
+            x_dim = 3 if image_format == 'zyx' else 1
+            z_dim = 1 if image_format == 'zyx' else 3
+            y_dim = 2
+
+        self.x_dim = x_dim
+        self.y_dim = y_dim
+        self.z_dim = z_dim
+
+        z_dim_3d = 0 if image_format == 'zyx' else 2
+        self.z_dim_3d = z_dim_3d
+        self.image_format = image_format
+
+    def _place_block_z(self,
+                       pred_block,
+                       pred_image,
+                       start_idx,
+                       end_idx):
+        """Place the current block prediction in the larger vol
+
+        :param np.array pred_block: current prediction block
+        :param np.array pred_image: full 3D prediction image with zeros
+        :param int start_idx: start slice of pred_block
+        :param int end_idx: end slice of pred block
+        :return np.array pred_image: with pred_block placed accordingly
```

Comment:
Do you need to return pred_image when it's passed into the function by reference and mutated in place?